### DISCUSSION TITLE
How to refresh the diagnose

### DISCUSSION THREAD

**Discussion by Decodetalkers** [ebkalderon/tower-lsp#348] [Q&A] [answered]
I publish_diagnostics to the editor
but I want to refresh these diagnose on the next change. How to refresh these diagnose?

**Comment by ratmice** [comment, marked as answer]
From what i've observed, the next call to publish_diagnostics for the file should refresh them, if you have no diagnostics to send, just send an empty vector for the `diags` parameter, and it should then clear the existing diagnostics.

**Reply by Decodetalkers** [reply]
Thanks very much
